Add Oracle OKE support for NVIDIA NIM

akuppala21 opened the pull request:
This PR adds support for deploying NVIDIA NIM LLM containers on Oracle Cloud Infrastructure (OCI) using Oracle Kubernetes Engine (OKE).

curt-lockhart reviewed (approved):
README update looks good!

juanpabloguerra16 reviewed (changes requested):
a few observations:

- why do we need a read me at the Oracle folder level with specific instructions? I'd suggest to place the readme in a platform specific sample, e.g. oracle/oke
- why are you using a deployment object to deploy the NIM? Why not use helm chart? That's the recommended way to deploy nims on K8S envs

akuppala21 commented:
Thanks @juanpabloguerra16 — I’ve updated the PR to follow your feedback:
	•Moved the README to cloud-service-providers/oracle/oke/
	•Removed all deployment YAMLs and switched to the shared helm/nim-llm chart
	•Included an OCI-specific values.yaml and instructions aligned with the other CSPs (Azure, GCP, AWS)

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
````diff
@@ -0,0 +1,202 @@
+
+### 1. Create a Virtual Cloud Network (VCN)
+
+- Public Subnet: For OKE worker nodes  
+- Private Subnet (optional): For internal services  
+- NAT Gateway or Internet Gateway (if using public IPs)  
+- Ensure ports `443` and `8000` are open in your NSG or security list (for testing)
+
+---
+
+### 2. Create a NAT Gateway (Recommended)
+
+```bash
+oci network route-rule add --route-table-id <ROUTE_TABLE_OCID> \
+  --destination 0.0.0.0/0 \
+  --network-entity-id <NAT_GATEWAY_OCID>
````
does this command cover the public subnet creation or only a routing rule?

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
````diff
@@ -0,0 +1,202 @@
+  --from-literal=NGC_API_KEY=$NGC_API_KEY
+```
+
+#### Sample `values.yaml`:
````
it's not clear if this `values.yaml` needs to be created or modified in the chart

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
```diff
@@ -0,0 +1,202 @@
+
+---
+
+### 4. Add Node Pool (A100)
```
do we need A100s to run an 8b model? May be an overkill, how about A10s?

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
```diff
@@ -0,0 +1,295 @@
+- Helm installed locally  
+- NVIDIA NGC API key from [ngc.nvidia.com](https://ngc.nvidia.com)  
+- Access to private NIM Helm charts via NGC  
+- IAM permissions to create:
+  - VCNs, Subnets, NAT/Internet Gateways, Route Tables
+  - OKE Clusters and Node Pools
+  - Secrets and Vaults
+  - Instance Configuration and Pools (for GPU node provisioning)
```
not necessarily a blocker to merge this PR but just an improvement - may be good to list the specific iam roles that include these permissions rather than stating them in general

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
````diff
@@ -0,0 +1,295 @@
+```bash
+oci network security-list update \
+  --security-list-id <SECURITY_LIST_OCID> \
+  --ingress-security-rules '[{"protocol":"6","source":"0.0.0.0/0","tcpOptions":{"destinationPortRange":{"min":8000,"max":8000}}, "isStateless":false},{"protocol":"6","source":"0.0.0.0/0","tcpOptions":{"destinationPortRange":{"min":443,"max":443}}, "isStateless":false}]'
````
why is opening ingress to 0.0.0.0 required? This might not be recommended

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
```diff
@@ -0,0 +1,295 @@
+| A100 (SXM)                  | 80 GB  | SXM       |
+| L40S                        | 48 GB  | PCIe      |
+| A10G                        | 24 GB  | PCIe      |
+| RTX 6000 Ada                | 48 GB  | PCIe      |
+| GeForce RTX 5090            | 32 GB  | PCIe      |
+| GeForce RTX 4090            | 24 GB  | PCIe      |
+| GeForce RTX 4080            | 16 GB  | PCIe      |
```
are these gpus available on OCI? Im not sure if it makes sense to include them

juanpabloguerra16 reviewed (commented):
I don't have further observations. I will defer final approval to Richard as there's some OCI specific commands outside of my expertise.

richardincyberspace commented:
Overall looks good.  Should we consider including a sample configuration with persistent storage?  With the default OCI GPU node configuration, the node could run out of disk space for any of the larger models.

akuppala21 commented:
Made the changes in values.yaml and readme.md.

juanpabloguerra16 commented on cloud-service-providers/oracle/README.md:
```diff
@@ -0,0 +1,331 @@
+
+---
+
+## 📝 Create or Edit `values.yaml`
+
+You must create or modify this file to match your deployment.
```
I'd recommend to bemore specific as tho how to get to this file, would they need  to:

cd charts/nim-llm 

then you'll find the file, right?

akuppala21 commented:
Resolved issues with capacity issues nodes and on demand nodes. Now handles both types and more detailed readme.md/values.yaml

Tested E2E
